src/test/unit_tests/test_s3_ops.js

```diff
@@ -94,51 +95,63 @@ mocha.describe('s3_ops', function() {
             if (bucket_type === "regular") {
                 await s3.createBucket({ Bucket: bucket_name }).promise();
                 await s3.createBucket({ Bucket: source_bucket }).promise();
-            } else if (SKIP_TEST) {
-                this.skip(); // eslint-disable-line no-invalid-this
             } else {
+                /**
+                 * NOTE: Remote buckets are not created in the test. It is assumed that
+                 * a. The buckets exist
+                 * b. The buckets are empty
+                 * Following buckets need to exist on the remote endpoint:
+                 *
+                 * remote-s3-ops-bucket
+                 * remote-s3-ops-bucket-caching
+                 * remote-s3-ops-source
+                 * remote-s3-ops-source-caching
+                 */
+                REMOTE_SOURCE_BUCKET = caching ? SOURCE_BUCKET + '-caching' : SOURCE_BUCKET;
+                REMOTE_TARGET_BUCKET = caching ? TARGET_BUCKET + '-caching' : TARGET_BUCKET;
+
                 const read_resources = [RESOURCE_NAME];
                 const write_resource = RESOURCE_NAME;
                 const read_resources1 = [RESOURCE_NAME_SOURCE];
                 const write_resource1 = RESOURCE_NAME_SOURCE;
-                const ENDPOINT = process.env.ENDPOINT ? process.env.ENDPOINT : 'https://s3.amazonaws.com';
-                const ENDPOINT_TYPE = process.env.ENDPOINT_TYPE ? process.env.ENDPOINT_TYPE : 'AWS';
+                const ENDPOINT = USE_LOCAL_ENDPOINT ? coretest.get_https_address() : process.env.ENDPOINT;
+                const ENDPOINT_TYPE = USE_LOCAL_ENDPOINT ? 'AWS' : process.env.ENDPOINT_TYPE;
+                const AWS_ACCESS_KEY_ID = USE_LOCAL_ENDPOINT ? s3.config.accessKeyId : process.env.AWS_ACCESS_KEY_ID;
+                const AWS_SECRET_ACCESS_KEY = USE_LOCAL_ENDPOINT ? s3.config.secretAccessKey : process.env.AWS_SECRET_ACCESS_KEY;
+
+                if (USE_LOCAL_ENDPOINT) {
+                    process.env.NODE_TLS_REJECT_UNAUTHORIZED = 0;
```

Comment:
Why is that needed in such a global way? 
For endpoints with self-signed certificates we usually set the option of a local s3 sdk client with `agent: http_utils.get_default_agent(endpoint)` or `agent: http_uttils.get_unsecured_agent(endpoint)`, for example see here:
https://github.com/noobaa/noobaa-core/blob/74b79637d8a9a1dca2baf51598860579711a5c6b/src/agent/block_store_services/block_store_s3.js#L32-L62

Reply:
I was running into errors such as `NetworkingError: self signed certificate`

At: https://github.com/noobaa/noobaa-core/blob/a075e9bdc8cced6e4b517e6161fe5aed27ff4052/src/test/unit_tests/test_s3_ops.js#L51, a http endpoint is created  (This is where I was adding get_unsecured_agent/get_default_agent)

When creating an external connection, I added an `https` connection (https://github.com/noobaa/noobaa-core/blob/a075e9bdc8cced6e4b517e6161fe5aed27ff4052/src/test/unit_tests/test_s3_ops.js#L120)

When a namespace bucket is created using this connection, I was getting (a PUT object to the namespace bucket) either a 
* `self signed certificate` error if I used `https` for external connection endpoint -or-
* `Error: Non-file stream objects are not supported with SigV4` if I used `http` for the external endpoint connection.

Setting `process.env.NODE_TLS_REJECT_UNAUTHORIZED` I was able to get the tests to run. 
I did try out the same sequence in the management console and did not encounter any issues.

Reply:
try to update this line:
https://github.com/noobaa/noobaa-core/blob/7c8eedea337f7fbca975f720ace1927065e46692/src/test/unit_tests/test_s3_ops.js#L56

to:
```
agent: http_utils.get_unsecured_agent(endpoint) 
```

Reply:
yes, I tried it. I get: `NetworkingError: self signed certificate`

So the error is triggered at: https://github.com/noobaa/noobaa-core/blob/a075e9bdc8cced6e4b517e6161fe5aed27ff4052/src/test/unit_tests/test_s3_ops.js#L189

for a PUT object to the `namespace`/`caching namespace` bucket. 

The namespace bucket is created with the external connection at `https://github.com/noobaa/noobaa-core/blob/a075e9bdc8cced6e4b517e6161fe5aed27ff4052/src/test/unit_tests/test_s3_ops.js#L131`.

When the external endpoint is set to IBM COS, the write to the namespace bucket is successful. This only happens when the endpoint is the noobaa endpoint (`coretest.get_https_address()`)

Reply:
Oh ok, but I see you replaced it in latest update. Is that working now?